packages/check/match.js

```diff
@@ -368,9 +368,9 @@ var testSubtree = function (value, pattern) {
   });
 
   //XXX: replace with underscore's _.allKeys if Meteor updates underscore to 1.8+ (or lodash)
-  var allKeys = function(obj){
+  function allKeys(obj) {
     var keys = [];
-    if (_.isObject(obj)){
+    if (obj) {
       for (var key in obj) keys.push(key);
     }
     return keys;
```

Comment:
So… this is an interesting case, because the loop over the keys below could just be a `for`-`in` loop, and you wouldn't need this `allKeys` function at all:
```js
for (var key in value) {
  var subValue = value[key];
  ...
```
This almost works, but it fails if `value` is not something object-like.

To fix that problem, you can just coerce it with the `Object` constructor:
```js
for (var key in Object(value)) {
  var subValue = value[key];
  ...
```
This works because `Object(value) === value` if it's object-like, but `Object(value)` evaluates to a new empty object if it's anything else. In those cases the loop body will never execute, so accessing `value[key]` there should be safe.

In fact, underscore and lodash implement `_.isObject` with `return value === Object(value)`, demonstrating the safety of this approach.